### PR TITLE
Pre release fixes pack

### PR DESCRIPTION
**Pull request by jacekkolasa**
a pack of fixes for first dashboard release (beta). when reviewing, i think it's better to do that commit-by-commit, because otherwise the changes are not related to each other

### CODE REVIEW

File: src/hooks/use-show-value-outside.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    const { showValueOf } = attributes
+    // showValueOf will be undefined if not used, but additional isEmpty check can prevent
+    // regression performance issue in the future
+    if (!showValueOf || isEmpty(showValueOf)) {
+      return
+    }
+    const dimensionNames = chartData.dimension_names
+    const dimensionIds = chartData.dimension_ids
+    dimensionNames.forEach((dimensionName, i) => {
+      const userElementId = showValueOf[`show-value-of-${dimensionName.toLowerCase()}`]
```
Comment (Azein) [inline src/hooks/use-show-value-outside.ts]: Could be an external util, maybe this level of detail how we call the key isn't so important.

File: src/hooks/use-show-value-outside.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        || showValueOf[`show-value-of-${dimensionIds[i].toLowerCase()}-at`]
+
+      // if element is not found, just add null
+      showValueAttributesNodes.current = showValueAttributesNodes.current.concat(
+        document.getElementById(userElementId),
+      )
+    })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []) // do only once on purpose
```
Comment (Azein) [inline src/hooks/use-show-value-outside.ts]: I think, react-use has an effect for this, which doesn't require eslint disable / additional commenting. Like useDidMount or something.

File: src/hooks/use-show-value-outside.ts (hunk 1 above)
Comment (jacekkolasa) [reply src/hooks/use-show-value-outside.ts]: do you mean that `show-value-of-${dimension}-at` is too descriptive? This is the exact name of an attribute from custom dashboards

File: src/hooks/use-show-value-outside.ts (hunk 2 above)
Comment (jacekkolasa) [reply src/hooks/use-show-value-outside.ts]: good idea, will do that now

File: src/hooks/use-show-value-outside.ts (hunk 1 above)
Comment (Azein) [reply src/hooks/use-show-value-outside.ts]: Well, technically of course this is needed, I'm saying that probably it's not required to see the exact string templates when reading the code there.

**Pull request merged**